pkg/util/progressdots/progressdots.go

```diff
@@ -21,18 +21,27 @@ import (
 	"time"
 )
 
-const DefaultInterval = 2 * time.Second
+const (
+	DefaultInterval   = 2 * time.Second
+	DefaultDotCounter = 10
+)
 
 type ProgressDots struct {
-	interval time.Duration
-	handler  chan bool
+	interval   time.Duration
+	dotCounter int
+	handler    chan struct{}
 }
 
 // NewProgressDots creates the channel to handle progress dots
-func New() *ProgressDots {
+func New(dotCounterOptional ...int) *ProgressDots {
+	dotCounter := DefaultDotCounter
+	if len(dotCounterOptional) > 0 {
+		dotCounter = dotCounterOptional[0]
+	}
 	return &ProgressDots{
-		interval: DefaultInterval,
-		handler:  make(chan bool),
+		interval:   DefaultInterval,
+		dotCounter: dotCounter,
+		handler:    make(chan struct{}),
```

Comment:
`handler: make(chan bool);` makes ProgressDots easier to re-use with a function `SetHandler` to allow for syncing with an external handler

```go
func main() {
    done := make(chan bool)
    
    p := progressDots.New()
    p.SetHandler(done)
    p.Start()
    
    <- done
```

Reply:
@gbraad Currently we don't expose a sethandler functionality for progressDots but in future we require then I do agree on this change.